packages/calcite-components/src/components/combobox/combobox.e2e.ts

```diff
@@ -1805,42 +1782,78 @@ describe("calcite-combobox", () => {
-        await page.waitForChanges();
-        expect(await page.evaluate(() => document.activeElement.id)).toBe("button");
-        expect(await combobox.getProperty("value")).toBe("");
-      }
+      await page.mouse.move(10, 2 * comboboxRect.bottom);
+      await page.mouse.down();
+      await page.waitForChanges();
+      await page.mouse.up();
+      await page.waitForChanges();
+      expect(await page.evaluate(() => document.activeElement.id)).not.toBe("myCombobox");
+      expect(await inputEl.getProperty("value")).toBe("");
+      expect(await combobox.getProperty("value")).toBe(allowCustomValues ? "asdf" : "");
     }
 
-    it("should clear input value in single selectionMode", async () => {
-      await assertClickOutside("single");
-    });
+    selectionModes.forEach((mode) => {
+      it(`should clear input value when selectionMode=${mode} `, async () => {
+        await assertClickOutside(mode);
+      });
 
-    it("should not clear input value in single selectionMode with allowCustomValues", async () => {
-      await assertClickOutside("single", true);
+      it(`should not clear input value when selectionMode=${mode} with allowCustomValues`, async () => {
+        await assertClickOutside(mode, true);
+      });
     });
+  });
+
+  describe("custom input value on blur using keyboard", () => {
+    let page: E2EPage;
 
-    it("should clear input value in multiple selectionMode", async () => {
-      await assertClickOutside();
+    beforeEach(async () => {
+      page = await newE2EPage();
+      await page.setContent(
+        html`
+          <calcite-combobox id="myCombobox">
+            <calcite-combobox-item id="one" value="one" text-label="one"></calcite-combobox-item>
+            <calcite-combobox-item id="two" value="two" text-label="two"></calcite-combobox-item>
+          </calcite-combobox>
+        `
+      );
     });
 
-    it("should not clear input value in multiple selectionMode with allowCustomValues", async () => {
-      await assertClickOutside("multiple", true);
+    async function clearInputValueOnBlur(selectionMode = "multiple", allowCustomValues = false): Promise<void> {
+      const combobox = await page.find("calcite-combobox");
+      combobox.setProperty("selectionMode", selectionMode);
+      combobox.setProperty("allowCustomValues", allowCustomValues);
+      const inputEl = await page.find(`#myCombobox >>> input`);
+      await inputEl.focus();
+      await page.waitForChanges();
+      expect(await page.evaluate(() => document.activeElement.id)).toBe("myCombobox");
+      await page.keyboard.type("asdf");
+      await page.waitForChanges();
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      expect(await inputEl.getProperty("value")).toBe("");
+      expect(await combobox.getProperty("value")).toBe(allowCustomValues ? "asdf" : "");
+    }
+    selectionModes.forEach((mode) => {
```

Comment:
Nitpick: for readability, can you add a newline above this to make it easier to scan?